src/CoWSwapEthFlow.sol

```diff
@@ -116,13 +116,44 @@ contract CoWSwapEthFlow is
             address(this),
             cowSwapOrder.validTo
         );
-        uint256 freedAmount = cowSwapOrder.sellAmount -
-            cowSwapSettlement.filledAmount(orderUid);
+        uint256 filledAmount = cowSwapSettlement.filledAmount(orderUid);
+
+        // This comment argues that a CoW Swap trader does not pay more fees if a partially fillable order is
+        // (partially) settled in multiple batches rather than in one single batch of the combined size.
+        // This also mean that we can refund the user assuming the worst case of settling the filled amount in a single
+        // batch without risking giving out more funds than available in the contract because of rounding issues.
+        // A CoW Swap trader is always charged exactly the amount of fees that is proportional to the filled amount
+        // rounded down to the smaller integer. The code is here:
+        // https://github.com/cowprotocol/contracts/blob/d4e0fcd58367907bf1aff54d182222eeaee793dd/src/contracts/GPv2Settlement.sol#L385-L387
+        // Our original statement is equivalent to `floor(a/c) + floor(b/c) ≤ floor((a+b)/c)`. Writing a and b in terms
```

Comment:
You could write it here more explicit:

For an order that was partially filled two times with two fee substraction `floor(a/c)`  `floor(b/c)`, the contract will also not payout too much funds, as `floor(a/c) + floor(b/c) ≤ floor((a+b)/c)` holds: Writing a and b.

At least I needed to think shortly about it...

Reply:
I updated the comment, do you think the new changes help?

Reply:
Yes, I like it!